FinancialTransactionsApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -118,5 +102,48 @@ public async Task UpdateAsync(Transaction transaction)
         {
             await _dynamoDbContext.SaveAsync(transaction.ToDatabase()).ConfigureAwait(false);
         }
+
+
+        public async Task<List<Transaction>> GetTransactionsAsync(Guid targetId, string transactionType, DateTime? startDate, DateTime? endDate)
```

Comment:
I think we should always page results and maybe have a default page size that can be overwritten - otherwise, we risk not returning all records (As we've hit payload limits) if there are a lot of transactions